cloudify_cli/tests/commands/test_events.py

```diff
@@ -79,10 +80,8 @@ def _mock_executions_get(self, execution_id):
 
     def _mock_events_get(self, execution_id, from_event=0,
                          batch_size=100, include_logs=False):
-        if from_event >= len(self.events):
-            return [], len(self.events)
-        until_event = min(from_event + batch_size, len(self.events))
-        return self.events[from_event:until_event], len(self.events)
+        events = self._get_events_before(time.time())
+        return events[from_event:], len(events)
```

Comment:
Good catch! Fixed now. (the `min` is redundant since slicing beyond the bounds would just return only up to the end of the list)

Otherwise I think the function should be understandable? In the previous/threaded version, one thread was generating the events, and another was running the function under test that polled for events. This version pre-generates all the events, and only returns those that come with a timestamp smaller than "now".

Btw the function under test is still doing a `sleep()`, as it was before.
We could also mock out the `time.sleep` and `time.time` calls so that then the tests would run instantly, rather than in 12 seconds :)
But I didn't want to add another behaviour change to this PR